### ISSUE TITLE
Kinetics-600 inference

### ISSUE THREAD

**Issue by Polycoriander** [kwsong0113/diffusion-forcing-transformer#1]
Hello, your work is excellent, and I have learned a lot from it. I have a question regarding my inference on the Kinetics-600 model. I noticed that most of the generated videos are static, essentially showing the same unchanging image. Could you please help me determine if there is an issue with my script settings? Here is my script:
`python -m main +name=single_image_to_short dataset=kinetics_600 algorithm=dfot_video experiment=video_generation @DiT/XL load=DFoT_K600.ckpt 'experiment.tasks=[validation]' experiment.validation.data.shuffle=True dataset.context_length=1 dataset.frame_skip=1 dataset.n_frames=17 experiment.validation.batch_size=1 algorithm.tasks.prediction.history_guidance.name=vanilla +algorithm.tasks.prediction.history_guidance.guidance_scale=4.0`

**Comment by buoyancy99** [comment]
As we mentioned in the paper, **vanilla history guidance** leads to static image when guidance scale is high, which can be addressed using fractional guidance. We will release a kinetics inference command shortly for you to run it!

**Comment by kwsong0113** [comment]
### Updates on Kinetics-600 video generation
- The original Kinetics-600 videos already contain minimal motion. As mentioned earlier, using vanilla history guidance can make the generated videos appear more static. Therefore, we recommend using **_smaller guidance scales of 1.5 or 3_**, as employed in our paper. To adjust this yourself, you can start with a guidance scale of 1 (no history guidance) and gradually increase it.
-  In our paper, we **_set `dataset.context_length` to 5_**, following the standard Kinetics-600 benchmark. If you aim to precisely replicate our results, you might consider following this setting.
-  You can generate **longer 65-frame videos** on Kinetics-600 using a sliding window rollout (Section 6.3 in our paper), again using guidance scales of 1.5 or 3. The reproducible commands are available in [this wiki](https://github.com/kwsong0113/diffusion-forcing-transformer/wiki/Reproducing#sec-63-kinetics-600-history-guidance-experiment). You can choose fractional history guidance (HG-f), with `freq_scale` of 0.6 or 0.8, to make videos more dynamic!

**Comment by Polycoriander** [comment]
/

> As we mentioned in the paper, **vanilla history guidance** leads to static image when guidance scale is high, which can be addressed using fractional guidance. We will release a kinetics inference command in a day or two for you to reproduce our result!

Thank you very much!

**Comment by Polycoriander** [comment]
> ### Updates on Kinetics-600 video generation
> * The original Kinetics-600 videos already contain minimal motion. As mentioned earlier, using vanilla history guidance can make the generated videos appear more static. Therefore, we recommend using **_smaller guidance scales of 1.5 or 3_**, as employed in our paper. To tune this yourself, you can start with a guidance scale of 1 (no history guidance) and gradually increase it.
> * In our paper, we **_set `dataset.context_length` to 5_**, following the standard Kinetics-600 benchmark. If you aim to precisely replicate our results, you might consider following this setting.
> * You can generate **longer 65-frame videos** on Kinetics-600 using a sliding window rollout (Section 6.3 in our paper), again using guidance scales of 1.5 or 3. The reproducible commands are available in [this wiki](https://github.com/kwsong0113/diffusion-forcing-transformer/wiki/Reproducing#sec-63-kinetics-600-history-guidance-experiment). Additionally, you can choose fractional history guidance (HG-f), with `freq_scale` of 0.6 or 0.8, to make videos more dynamic!

Thank you for your explanation, I will try it out.

**Comment by Polycoriander** [comment]
> ### Updates on Kinetics-600 video generation
> * The original Kinetics-600 videos already contain minimal motion. As mentioned earlier, using vanilla history guidance can make the generated videos appear more static. Therefore, we recommend using **_smaller guidance scales of 1.5 or 3_**, as employed in our paper. To tune this yourself, you can start with a guidance scale of 1 (no history guidance) and gradually increase it.
> * In our paper, we **_set `dataset.context_length` to 5_**, following the standard Kinetics-600 benchmark. If you aim to precisely replicate our results, you might consider following this setting.
> * You can generate **longer 65-frame videos** on Kinetics-600 using a sliding window rollout (Section 6.3 in our paper), again using guidance scales of 1.5 or 3. The reproducible commands are available in [this wiki](https://github.com/kwsong0113/diffusion-forcing-transformer/wiki/Reproducing#sec-63-kinetics-600-history-guidance-experiment). Additionally, you can choose fractional history guidance (HG-f), with `freq_scale` of 0.6 or 0.8, to make videos more dynamic!

Hello, I have another question to ask. Does the DFoT model necessarily require images and cannot generate videos from condition alone？

**Comment by kwsong0113** [comment]
@Polycoriander You can generate a video from any number of images, even with no images (i.e., ⁠dataset.context_length=0 is possible). Note that the quality of the videos improves with more images provided, as this makes the task easier.

**Comment by Polycoriander** [comment]
> [@Polycoriander](https://github.com/Polycoriander) You can generate a video from any number of images, even with no images (i.e., ⁠dataset.context_length=0 is possible). Note that the quality of the videos improves with more images provided, as this makes the task easier.

Thank you!

**Comment by Polycoriander** [comment]
> [@Polycoriander](https://github.com/Polycoriander) You can generate a video from any number of images, even with no images (i.e., ⁠dataset.context_length=0 is possible). Note that the quality of the videos improves with more images provided, as this makes the task easier.

Hello,

When I use your pretrained model pretrained:DFoT_K600.ckpt for inference and set dataset.context_length=0, I can generate videos normally. Therefore, I tried to follow your training script to train the model from scratch using a portion of the K600 dataset. The training loss is as follows, and the denoised images in denoising_vis are very close to the ground truth (gt). However, when I set dataset.context_length=0 for prediction, the results are very poor as follows.

Could you help me look into this issue? Or do you have any previous experience with similar problems?

![Image](https://github.com/user-attachments/assets/cb7e0fcc-db57-484c-8631-06424b5327d7)
![Image](https://github.com/user-attachments/assets/477d5388-de23-4c02-918b-33cff57ae025)

**Comment by buoyancy99** [comment]
Depending on the model size, unconditioned generation is extremely hard in unconditional case - even non of the video foundation models would be able to generate well without text prompts, due to the diversity of the datasets
